### PR TITLE
CDAP-20410: Add error subclasses for specific source control errors

### PR DESCRIPTION
**Pull request by samdgupi**
Added subclasses for SourceControlException for specific class of errors
1. GitOperationException in case of failure in git operations like getting filehash, commit, push, pull
2. ConfigFileWriteException in case we fail to write config file

### CODE REVIEW

File: cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/ConfigFileWriteException.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.sourcecontrol;
+
+/**
+ * Exception thrown when source control operation fails due to underlying filesystem IO exception
+ * while reading/writing cloned repository
```
Comment (de-lan) [inline cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/ConfigFileWriteException.java]: nit: does it include the `reading`?

File: cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManager.java [hunk 2]
```diff
@@ -363,8 +363,7 @@ public String getFileHash(final Path relativePath, final String commitHash)
 
   @Nullable
   private String getFileHash(final Path relativePath, final RevCommit commit)
-      throws IOException,
-      GitAPIException {
+      throws IOException{
```
Comment (de-lan) [inline cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManager.java]: nit: should be an empty space before `{`

File: cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/SourceControlException.java [hunk 3]
```diff
@@ -14,14 +14,13 @@
...
 
-package io.cdap.cdap.sourcecontrol.operationrunner;
+package io.cdap.cdap.sourcecontrol;
 
 /**
  * Exception thrown when source control operation fails in runner.
  * Encapsulates some underlying exceptions.
  * Should be subclassed for common root-causes
  */
-// TODO https://cdap.atlassian.net/browse/CDAP-20410 Classify and make suberrors based on root causes
```
Comment (de-lan) [inline cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/SourceControlException.java:24]: 👍

**Review by de-lan — APPROVED** [review]
some nit comments

File: cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/ConfigFileWriteException.java (hunk 1 above)
Comment (samdgupi) [reply cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/ConfigFileWriteException.java]: fixed

File: cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManager.java (hunk 2 above)
Comment (samdgupi) [reply cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManager.java]: fixed

**Review by tivv — APPROVED** [review]

**Pull request merged**